x/pairing/keeper/msg_server_relay_payment_test.go

```diff
@@ -482,81 +479,85 @@ func TestEpochPaymentDeletion(t *testing.T) {
 	require.Equal(t, 0, len(res.EpochPayments))
 }
 
-// Test that after the consumer uses some CU it's updated in its project and subscription
-func TestCuUsageInProjectsAndSubscription(t *testing.T) {
-	ts := newTester(t)
-	ts.SetupAccounts(0, 0, 2)    // 0 sub, 0 adm, 2 dev
-	ts.setupForPayments(1, 1, 0) // 1 provider, 1 client, default providers-to-pair
-
-	dev1Acct, dev1Addr := ts.Account("dev1")
-	_, dev2Addr := ts.Account("dev2")
-
-	_, client1Addr := ts.GetAccount(common.CONSUMER, 0)
-	providerAcct, providerAddr := ts.GetAccount(common.PROVIDER, 0)
-
-	projectData := projecttypes.ProjectData{
-		Name:    "proj1",
-		Enabled: true,
-		ProjectKeys: []projecttypes.ProjectKey{
-			projecttypes.ProjectDeveloperKey(dev1Addr),
-		},
-		Policy: &planstypes.Policy{
-			GeolocationProfile: int32(1),
-			MaxProvidersToPair: 3,
-			TotalCuLimit:       1000,
-			EpochCuLimit:       100,
-		},
-	}
-	err := ts.TxSubscriptionAddProject(client1Addr, projectData)
-	require.Nil(t, err)
-
-	projectData.Name = "proj2"
-	projectData.ProjectKeys = []projecttypes.ProjectKey{
-		projecttypes.ProjectDeveloperKey(dev2Addr),
-	}
-	err = ts.TxSubscriptionAddProject(client1Addr, projectData)
-	require.Nil(t, err)
-
-	ts.AdvanceEpoch()
-
-	qos := types.QualityOfServiceReport{
-		Latency:      sdk.OneDec(),
-		Availability: sdk.OneDec(),
-		Sync:         sdk.OneDec(),
-	}
-
-	relaySession := ts.newRelaySession(providerAddr, 0, 1, ts.BlockHeight(), 0)
-	relaySession.QosReport = &qos
-	relaySession.Sig, err = sigs.Sign(dev1Acct.SK, *relaySession)
-	require.Nil(t, err)
-
-	// Request payment (helper validates the balances and verifies expected errors through valid)
-	relayPaymentMessage := types.MsgRelayPayment{
-		Creator: providerAddr,
-		Relays:  slices.Slice(relaySession),
-	}
-
-	ts.payAndVerifyBalance(relayPaymentMessage, dev1Acct.Addr, providerAcct.Addr, true, true, 100)
-
-	sub, err := ts.QuerySubscriptionCurrent(client1Addr)
-	require.Nil(t, err)
-	require.True(t, sub.Sub != nil)
-
-	proj1, err := ts.QueryProjectDeveloper(dev1Addr)
-	require.Nil(t, err)
-	require.Equal(t, sub.Sub.MonthCuTotal-sub.Sub.MonthCuLeft, proj1.Project.UsedCu)
-
-	proj2, err := ts.QueryProjectDeveloper(dev2Addr)
-	require.Nil(t, err)
-	require.NotEqual(t, sub.Sub.MonthCuTotal-sub.Sub.MonthCuLeft, proj2.Project.UsedCu)
-}
+// TODO: not sure how to fix this test. The test checks that the monthly CU is updated correctly on the
```

Comment:
you need to pass a month, then check for the previous month, right when it's about to be delted

Reply:
Note that this test checks the CU count in the project/subscription (and not tracked CU, which is tested elsewhere). If I do as you say, I won't be able to check the CU count since it'll reset.
To fix the test, I simply sent the relay payment without advancing a month. The CU is counted in the relay payment TX, so there is no need to advance a month.

see https://github.com/lavanet/lava/commit/f98b64e5c6080a03eeb1e55549b3c12c3d3033a8